hazelcast/src/main/java/com/hazelcast/sql/SqlService.java

```diff
@@ -0,0 +1,147 @@
+ * located on the member that initiates the query. The engine extracts columns from a key and a value and then merges them
+ * into a single column set. In case the key and the value have columns with the same name, the key takes precedence.
+ * <p>
+ * Columns are extracted from objects as follows:
+ * <ul>
+ *     <li>For non-Portable objects, public getters and fields are used to populate the column list. For getters, the first
+ *     letter is converted to lower case. A getter takes precedence over a field in case of naming conflict
+ *     <li>For {@link Portable} objects, field names used in the {@link Portable#writePortable(PortableWriter)} method
+ *     are used to populate the column list
+ * </ul>
+ * The whole key and value objects could be accessed through a special fields {@code __key} and {@code this}, respectively. If
+ * key (value) object has fields, then the whole key (value) field is exposed as a normal field. Otherwise the field is hidden.
+ * Hidden fields can be accessed directly, but are not returned by {@code SELECT * FROM ...} queries.
+ * <p>
+ * If the member that initiates a query doesn't have local entries for the given IMap, the query fails.
+ * <p>
+ * Consider the following key/value model:
+ * <pre>
+ *     class PersonKey {
+ *         private long personId;
+ *         private long deptId;
+ *
+ *         public long getPersonId() { ... }
+ *         public long getDepartmentId() { ... }
+ *     }
+ *
+ *     class Person {
+ *         public String name;
+ *     }
+ * </pre>
+ * This model will be resolved to the following table columns:
+ * <ul>
+ *     <li>personId BIGINT
+ *     <li>departmentId BIGINT
+ *     <li>name VARCHAR
+ *     <li>__key OBJECT (hidden)
+ *     <li>this OBJECT (hidden)
+ * </ul>
+ * <h2>Consistency</h2>
+ * Results returned from IMap query are weakly consistent:
+ * <ul>
+ *     <li>If an entry was not updated during iteration, it is guaranteed to be returned exactly once
+ *     <li>If an entry was modified during iteration, it might be returned zero, one or several times
+ * </ul>
+ * <h1>Usage</h1>
+ * When a query is executed, an {@link SqlResult} is returned. You may get row iterator from the result. The result must be closed
+ * at the end. The code snippet below demonstrates a typical usage pattern:
+ * <pre>
+ *     HazelcastInstance instance = ...;
+ *
+ *     try (SqlResult result = instance.sql().query("SELECT * FROM person")) {
+ *         for (SqlRow row : result) {
+ *             long personId = row.getObject("personId");
+ *             String name = row.getObject("name");
+ *             ...
+ *         }
+ *     }
+ * </pre>
+ */
+@Beta
+public interface SqlService {
+    /**
+     * Convenient method to execute a distributed query with the given parameters.
+     * <p>
+     * Converts passed SQL string and parameters into an {@link SqlQuery} object and invokes {@link #query(SqlQuery)}.
+     *
+     * @param sql SQL string
+     * @param params parameters
```

Comment:
I'd add a few words to explain - parameters for what?

Reply:
Added explanation.